### PR TITLE
Added data schema registry and data schema deployment stack

### LINKED ISSUE #257: Review and finalise event schema
## Context

Current we are following EventBridge doc to authoring our event schema.

- https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events-structure.html
- https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-schema-create.html
  - Following example: `StepFunctionsExecutionStatusChange` ^^

We are also empirically observing from `Amazon Console > EventBridge > Schemas > AWS event schema registry` on various AWS built-in schemas.

We are also choosing OpenAPI format over JSONSchema as OpenAPI format is observed to be most use format in built-in registry.

These schema has `AWSEvent` envelope object that wrap the domain event such as `StepFunctionsExecutionStatusChange` or, our case `SequnceRunStateChange` and so on.

After the event schema is put into registry, we can download the corresponding code binding through EventBridge Console UI.
- https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-schema-code-bindings.html

## Forces

There are few concerns we observed so far. As follows.

- Validation: Is EventBridge validate against corresponding schema registry on those messages enter into the bus? Especially archive events, for example. 
- The `detail-type` is a string? Such that:
```
...
"detail-type": "StepFunctionsExecutionStatusChange"
...
```
- What is the use of AWS x-headers? e.g.
```
"AWSEvent": {
  ...
  "x-amazon-events-detail-type": "Step Functions Execution Status Change",
  "x-amazon-events-source": "aws.states",
  ...
```
- Case issue: see our [poll](https://umccr.slack.com/archives/C03ABJTSN7J/p1713416590826269) - favour `camelCase`
- However, note also the envelope level object use `PascalCase`; such as `AWSEvent` and `StepFunctionsExecutionStatusChange` in detail-type property.

## CloudEvent

Study CloudEvent spec (https://github.com/cloudevents/spec) which seem to be well thought-out design.

Its SDK seems to be supported more language toolchains.
https://github.com/cloudevents/spec?tab=readme-ov-file#sdks

Its primer case study article that perform analysis across cloud events status quo.
https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/primer.md

TL;DR example: _JSON Event Format_

https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/json-format.md

```
{
    "specversion" : "1.0",
    "type" : "com.example.someevent",
    "source" : "/mycontext",
    "subject": null,
    "id" : "C234-1234-1234",
    "time" : "2018-04-05T17:31:00Z",
    "comexampleextension1" : "value",
    "comexampleothervalue" : 5,
    "datacontenttype" : "application/json",
    "data" : {
        "appinfoA" : "abc",
        "appinfoB" : 123,
        "appinfoC" : true
    }
}
```

## Actions

- Investigate AWS EventBridge schema registry inner details mechanic
- Experiment CloudEvent
- Discuss with team on find out

### PR DESCRIPTION
**Pull request by victorskl**
* Updated corresponding README and guide

Resolves #257

### CODE REVIEW

**Comment by alexiswl** [comment]
Can we resolve https://github.com/umccr/orcabus/issues/301 first?

See https://github.com/umccr/orcabus/pull/303 as well

**Comment by alexiswl** [comment]
Saves needing to migrate schemas in a few days' time

**Comment by victorskl** [comment]
Yup, good idea. Let us regroup and resolve those one by one, today.

**Comment by victorskl** [comment]
As discussed for #301, #303 - Ray will resolve with #299. When 299 ready, we shall review Ray's PR over there.

File: docs/schemas/data/bsshicav2fastqcopymanager/PayloadDataReady.json
Comment (victorskl) [inline docs/schemas/data/bsshicav2fastqcopymanager/PayloadDataReady.json:1]: As noted; any artefact under `docs/schemas/data/` is under the full ownership (as well as a contract declaration; two-way bonds) of workflow execution services.

These particular contents adhere in this PR are just as examples for starter. 

Please feel free to discard, improve, modify, rename as appropriate by the service owner see-fit.

FYI @raylrui @alexiswl

**Comment by victorskl** [comment]
If you could review sooner and tick approve (or comment if any) - would be good. Then I will merge and follow up to change them JSON Schema.

**Review by raylrui — APPROVED** [review]
LGTM

**Review by alexiswl — APPROVED** [review]

**Review by reisingerf — APPROVED** [review]

**Pull request merged**